modules/drivers/redshift/test/metabase/driver/redshift_test.clj

```diff
@@ -192,3 +196,29 @@
                    (map
                     (partial into {})
                     (db/select [Field :name :database_type :base_type] :table_id table-id))))))))))
+
+(defn- assert-jdbc-url-fetch-size [db fetch-size]
+  (let [ds   (sql-jdbc.execute/datasource db)
+        conn (.getConnection ds)
+        md   (.getMetaData conn)
+        url  (.getURL md)]
+    (is (str/includes? url (str "defaultRowFetchSize=" fetch-size)))))
+
+(deftest test-jdbc-fetch-size
+  (testing "Redshift JDBC fetch size is set correctly in PreparedStatement"
+    (mt/test-driver :redshift
+      (assert-jdbc-url-fetch-size (mt/db) 5000) ; the default value should always be picked up if nothing is set
+      (with-redefs [env/env (assoc env/env :mb-redshift-fetch-size "14")]
```

Comment:
If it's defined with `defsetting` you have to use `mt/with-temporary-setting-values` to override the value, because it caches stuff internally

Reply:
Done.